packages/@ourworldindata/grapher/src/slopeCharts/SlopeChart.tsx

```diff
@@ -493,43 +511,163 @@ export class SlopeChart
+            showTextOutlines: true,
+            textOutlineColor: this.backgroundColor,
+            focusedSeriesNames: this.focusedSeriesNames,
+            onMouseOver: this.onLineLegendMouseOver,
+            onMouseLeave: this.onLineLegendMouseLeave,
+        }
+    }
+
+    @computed private get lineLegendPropsRight(): Partial<LineLegendProps> {
+        return {
+            xAnchor: "start",
+            fontWeight: 700,
+        }
+    }
+
+    @computed private get lineLegendPropsLeft(): Partial<LineLegendProps> {
+        return {
+            xAnchor: "end",
+            fontWeight: this.showSeriesNamesInLineLegendLeft ? 700 : undefined,
+            seriesSortedByImportance:
+                this.seriesSortedByImportanceForLineLegendLeft,
+        }
+    }
+
+    private formatValue(value: number): string {
+        return this.formatColumn.formatValueShortWithAbbreviations(value)
+    }
+
+    @computed get lineLegendMaxLevelLeft(): number {
+        if (!this.manager.showLegend) return 0
+
+        // can't use `lineLegendSeriesLeft` due to a circular dependency
+        const series = this.series.map((series) =>
+            this.constructSingleLineLegendSeries(
+                series,
+                (series) => series.start.value,
+                { showSeriesName: false }
+            )
         )
+
+        return LineLegend.maxLevel({
+            labelSeries: series,
+            ...this.lineLegendPropsCommon,
+            seriesSortedByImportance:
+                this.seriesSortedByImportanceForLineLegendLeft,
+            // not including `lineLegendPropsLeft` due to a circular dependency
+        })
     }
 
-    @computed get maxLineLegendWidth(): number {
-        // todo: copied from line legend (left padding, marker margin)
-        return Math.min(this.maxLabelWidth + 35 + 4, this.bounds.width / 3)
+    @computed get lineLegendWidthLeft(): number {
+        if (!this.manager.showLegend) return 0
+        return LineLegend.stableWidth({
+            labelSeries: this.lineLegendSeriesLeft,
+            ...this.lineLegendPropsCommon,
+            ...this.lineLegendPropsLeft,
+        })
+    }
+
+    @computed get lineLegendRight(): LineLegend | undefined {
+        if (!this.manager.showLegend) return undefined
+        return new LineLegend({
+            labelSeries: this.lineLegendSeriesRight,
+            ...this.lineLegendPropsCommon,
+            ...this.lineLegendPropsRight,
+        })
+    }
+
+    @computed get lineLegendWidthRight(): number {
+        return this.lineLegendRight?.stableWidth ?? 0
+    }
+
+    @computed get visibleLineLegendLabelsRight(): Set<SeriesName> {
+        return new Set(this.lineLegendRight?.visibleSeriesNames ?? [])
+    }
+
+    @computed get seriesSortedByImportanceForLineLegendLeft(): SeriesName[] {
+        return this.series
+            .map((s) => s.seriesName)
+            .toSorted((s1: SeriesName, s2: SeriesName): number => {
```

Comment:
While I love `toSorted`, and we actually do already polyfill it for older browsers, [it is not supported in Node 18.16](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted#browser_compatibility), which we currently use :(